clientconn.go

```diff
@@ -1229,6 +1163,22 @@ func (ac *addrConn) createTransport(addr resolver.Address, copts transport.Conne
 		return nil, err
 	}
 
+	if ac.dopts.reqHandshake == envconfig.RequireHandshakeOn {
+		select {
+		case <-prefaceTimer.C:
```

Comment:
We shouldn't still need a prefaceTimer, right?  We should be able to just use the normal connect deadline here I think.  And we can eliminate prefaceReceived, too, if we make NewClientTransport itself block until the preface is received.  That should simplify things even more.

Reply:
- prefaceTimer: removed
- onPrefaceReceipt: do you mind if I do that in a separate PR?